Mathlib/Algebra/Algebra/Defs.lean

```diff
@@ -410,4 +419,11 @@ end Semiring
 
 end Algebra
 
+@[norm_cast]
+theorem algebraMap.coe_smul (A B C : Type*) [SMul A B] [CommSemiring B] [Semiring C] [Algebra B C]
+    [SMul A C] [IsScalarTower A B C] (a : A) (b : B) : (a • b : B) = a • (b : C) := calc
+  ((a • b : B) : C) = (a • b) • 1 := Algebra.algebraMap_eq_smul_one _
+  _ = a • (b • 1) := smul_assoc ..
+  _ = a • (b : C) := congrArg _ (Algebra.algebraMap_eq_smul_one b).symm
```

Comment:
I think you can write this as `map_smul (Algebra.ofId B C) a b` or similar?

Reply:
`Algebra.restrictScalars` is defined in `Algebra/Algebra/Tower`, which comes after `Algebra/Algebra/Defs`